### PR TITLE
Add July Monthly Update Post

### PR DESCRIPTION
**Pull request by lauragift21**
Hi, @sonalkr132 and @deivid-rodriguez can you help look through this update to ensure I captured the work done in July properly. thanks!

### CODE REVIEW

**Review by sonalkr132 — APPROVED** [review]

File: _posts/2021-08-09-july-rubygems-updates.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+In addition to that, RubyGems saw several bug fixes and updates this month, some of which include the following:
+
+- investigated various RubyGems and Bundler issues on GitHub such as [#4717](https://github.com/rubygems/rubygems/pull/4717) and [#4719](https://github.com/rubygems/rubygems/pull/4719)
+- worked on the Bundler Version Locking RFC - [#29](https://github.com/rubygems/rfcs/pull/29).
+- merged an enhancement that will make setting the `RUBYGEMS_GEMDEPS` environment variable used for avoiding the need to use `bundle exec`.
+- worked on automating the process of reproducing builds for gems - [#3118](https://github.com/rubygems/rubygems/issues/3118)
+- improved performance of Bundler, shipped some bug fixes, and simplified Bundler internals.
+- fixed several outdated definition issues (detecting Gemfile changes over the lock file or not, to skip re-resolving them potentially. 
+-  fixed the `--conservative` flag to allow finer-grained bundle updates.
```
Comment (deivid-rodriguez) [inline _posts/2021-08-09-july-rubygems-updates.md:22]: My picky sight tells me there's a double leading space here :laughing:

```suggestion
- fixed the `--conservative` flag to allow finer-grained bundle updates. 
```

File: _posts/2021-08-09-july-rubygems-updates.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+In addition to that, RubyGems saw several bug fixes and updates this month, some of which include the following:
+
+- investigated various RubyGems and Bundler issues on GitHub such as [#4717](https://github.com/rubygems/rubygems/pull/4717) and [#4719](https://github.com/rubygems/rubygems/pull/4719)
+- worked on the Bundler Version Locking RFC - [#29](https://github.com/rubygems/rfcs/pull/29).
+- merged an enhancement that will make setting the `RUBYGEMS_GEMDEPS` environment variable used for avoiding the need to use `bundle exec`.
+- worked on automating the process of reproducing builds for gems - [#3118](https://github.com/rubygems/rubygems/issues/3118)
+- improved performance of Bundler, shipped some bug fixes, and simplified Bundler internals.
```
Comment (deivid-rodriguez) [inline _posts/2021-08-09-july-rubygems-updates.md:20]: I think this entry is essentially captured in the rest of the updates, so I would remove it.

File: _posts/2021-08-09-july-rubygems-updates.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+- fixed several outdated definition issues (detecting Gemfile changes over the lock file or not, to skip re-resolving them potentially.
```
Comment (deivid-rodriguez) [inline _posts/2021-08-09-july-rubygems-updates.md:21]: This entry read a bit weird to me, I tried to make it more clear

```suggestion
- fixed several outdated definition issues (detecting whether there are Gemfile changes over the lockfile or not, to potentially reuse the information in the lockfile and avoid having to resolve dependencies at all) 
```

File: _posts/2021-08-09-july-rubygems-updates.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+- investigated various RubyGems and Bundler issues on GitHub such as [#4717](https://github.com/rubygems/rubygems/pull/4717) and [#4719](https://github.com/rubygems/rubygems/pull/4719)
+- worked on the Bundler Version Locking RFC - [#29](https://github.com/rubygems/rfcs/pull/29).
+- merged an enhancement that will make setting the `RUBYGEMS_GEMDEPS` environment variable used for avoiding the need to use `bundle exec`.
+- worked on automating the process of reproducing builds for gems - [#3118](https://github.com/rubygems/rubygems/issues/3118)
+- improved performance of Bundler, shipped some bug fixes, and simplified Bundler internals.
+- fixed several outdated definition issues (detecting Gemfile changes over the lock file or not, to skip re-resolving them potentially. 
+-  fixed the `--conservative` flag to allow finer-grained bundle updates. 
+- worked on smoothing the transition to secure RubyGems sources fully and fixed the regressions we create as we proceed.
+
+In addition to making all these improvements, we achieved even faster Bundler/setup performance.
```
Comment (deivid-rodriguez) [inline _posts/2021-08-09-july-rubygems-updates.md:25]: While working on this changes we also simplified bundler internals and achieved faster `bundler/setup` performance.

**Review by deivid-rodriguez — COMMENTED** [review]
@lauragift21 Content looks good, just a made a few suggestions on wording.

**Review by deivid-rodriguez — APPROVED** [review]
:100:

**Pull request merged**